cpp/arcticdb/stream/index.hpp

```diff
@@ -296,10 +293,47 @@ class RowCountIndex : public BaseIndex<RowCountIndex> {
     static constexpr const char *name() { return "row_count"; }
 };
 
-using Index = std::variant<stream::TimeseriesIndex, stream::RowCountIndex, stream::TableIndex>;
+class EmptyIndex : public BaseIndex<EmptyIndex> {
+public:
+    using TypeDescTag = TypeDescriptorTag<DataTypeTag<DataType::EMPTYVAL>, DimensionTag<Dimension::Dim0>>;
+    static constexpr size_t field_count() {
+        return 0;
+    }
+
+    static constexpr IndexDescriptor::Type type() {
+        return IndexDescriptor::EMPTY;
+    }
+
+    static constexpr const char* name() {
+        return "empty";
+    }
+
+    static constexpr EmptyIndex default_index() {
+        return {};
+    }
+
+    [[nodiscard]] static IndexValue start_value_for_segment(const SegmentInMemory& segment) {
```

Comment:
Why the cast to a timestamp?

Reply:
The IndexValue::NumericType is typedefed to timestamp. I'll changing this to IndexValue::NumericType